src/main/java/com/zotov/edu/passportofficerestservice/repository/PassportsRepositoryCollections.java

```diff
@@ -0,0 +1,88 @@
+package com.zotov.edu.passportofficerestservice.repository;
+
+import com.zotov.edu.passportofficerestservice.repository.entity.Passport;
+import com.zotov.edu.passportofficerestservice.repository.entity.PassportState;
+import org.springframework.stereotype.Repository;
+
+import java.time.LocalDate;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Repository
+public class PassportsRepositoryCollections implements PassportsRepository {
+
+    Map<String, Passport> passports = new HashMap<>();
+
+    @Override
+    public Passport save(Passport passport) {
+        passports.put(passport.getNumber(), passport);
```

Comment:
If `passports` already contains an entity with such a key, it will be just overridden.
It might be better to check first whether the map contains such a key and throw an exception.
`openapi.yaml` also has to be adjusted as well to declare 409 status

Reply:
Fixed. Now it checks if a passport already exists in the data before updating the entity in a case when creating a new passport. Also, a new exception to handle a conflict has been added.